tests/api2/test_300_nfs.py

```diff
@@ -222,9 +222,13 @@ def set_immutable_state(path: str, want_immutable=True):
     '''
     Used by exportsd test
     '''
-    call('filesystem.set_immutable', want_immutable, path)
-    res = call('filesystem.is_immutable', '/etc/exports.d')
-    assert res is want_immutable, f"Expected mutable filesystem: {res}"
+    call('filesystem.set_zfs_attributes', {
+        'path': path,
+        'zfs_file_attributes': {'immutable': want_immutable}
+    })
+    is_immutable = 'IMMUTABLE' in call('filesystem.stat', '/etc/exports.d')['attributes']
+
+    assert is_immutable is want_immutable, f"Expected mutable filesystem: {is_immutable}"
```

Comment:
Text of message should really depend on value of `want_immutable`.  (Meh.)

Reply:
Correct!  That should be `want_immutable`

Reply:
Or
`assert is_immutable is want_immutable, f"Expected {'im' if want_immutable else ''}mutable filesystem: {is_immutable}"`

:-)

Reply:
Not necessary actually.   The failing assert will be output and the helpful 'expected' message.  What else is needed to see what failed on the test?